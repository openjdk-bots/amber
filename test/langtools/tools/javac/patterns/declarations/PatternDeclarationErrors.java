/*
 * @test /nodynamiccopyright/
 * @summary Verify error related to annotations and patterns
 * @enablePreview
 * @compile/fail/ref=PatternDeclarationErrors.out -XDrawDiagnostics -XDdev --should-stop=at=FLOW PatternDeclarationErrors.java
 */
public class PatternDeclarationErrors {
    public record D() {
        public pattern D(Object v1, Float out) {
            match D(10.0f);
        }

        public pattern D(Float out, Integer v1) {
            match D2(10.0f, 2);
        }

        public pattern D(Float out, Double v1) {
            match D(10.0f, "2");
        }

        public void D3(Float out, Integer v1) {
            match D3(10.0f, 2);
        }

        public pattern D(Float out, Character v1) {
            match D(10.0f, '2');
            System.out.println("unreachable");
        }
    }
}
