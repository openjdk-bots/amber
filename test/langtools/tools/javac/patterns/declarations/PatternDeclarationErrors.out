PatternDeclarationErrors.java:10:13: compiler.err.match.not.matching.pattern.declaration.signature
PatternDeclarationErrors.java:14:13: compiler.err.match.pattern.name.wrong
PatternDeclarationErrors.java:18:28: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.String, java.lang.Double)
PatternDeclarationErrors.java:22:13: compiler.err.match.outside.pattern.declaration
PatternDeclarationErrors.java:27:13: compiler.err.unreachable.stmt
- compiler.note.preview.filename: PatternDeclarationErrors.java, DEFAULT
- compiler.note.preview.recompile
5 errors